workflow/scripts/osemosys_global/transmission/data.py

```diff
@@ -1,51 +1,180 @@
+    as well as for the Nei Mongol region in China, assign custom OG regions as a 
+    best estimate for the most important transmission pathway. This pathway will have
+    all residual capacity assigned to it."""
+    df.loc[df['From'] == 'DUPLICATE', 
+           'From'] = df['TECHNOLOGY_gtd'].map(CUSTOM_TRN_BA_DICT_FROM)
+    
+    df.loc[df['To'] == 'DUPLICATE', 
+           'To'] = df['TECHNOLOGY_gtd'].map(CUSTOM_TRN_BA_DICT_TO)
+    
+    # Sort 'From' and 'To' columns alphabetically for custom entries. 
+    for idx,row in df.loc[df['To'] < df['From']].iterrows():
+        df.loc[df['TECHNOLOGY_gtd'] == row['TECHNOLOGY_gtd'], 
+               ['From', 'To']] = row['To'], row['From']
 
-    # If from column has length 9 then it's the 3:6 and 7:9 three chars plus XX
-    df.loc[df["From"].str.len() == 9, "From"] = (
-        df["From"].str[3:6] + df["From"].str[7:9])
+    df.set_index('TECHNOLOGY_gtd', inplace = True)
+    
+    # Create corrected TECHNOLOGY entry
+    df['TECHNOLOGY'] = 'TRN' + df['From'] + df['To']
 
-    # If to column has length 6 then it's the last three chars plus XX
-    df.loc[df["To"].str.len() == 6, "To"] = (df["To"].str[3:6] + "XX")
+    # Update original GTD df's with corrected entries
+    df_exist.set_index('TECHNOLOGY', inplace = True)
+    df_exist.update(df)
+    df_exist = df_exist.merge(df[['TECHNOLOGY']], left_index = True, 
+                              right_index = True).reset_index(drop = True)
+    df_exist = df_exist[df_exist['From'] != df_exist['To']]
+    
+    df_plan.set_index('TECHNOLOGY', inplace = True)
+    df_plan.update(df)
+    df_plan = df_plan.merge(df[['TECHNOLOGY']], left_index = True, 
+                            right_index = True).reset_index(drop = True)   
+    df_plan = df_plan[df_plan['From'] != df_plan['To']]
+    
+    # Group corrected GTD entries
+    df_exist = df_exist.groupby(['TECHNOLOGY', 'From', 'To']
+                                ).sum('VALUE').reset_index()
+    
+    df_plan = df_plan.groupby(['TECHNOLOGY', 'From', 'To', 'YEAR']
+                                ).sum('VALUE').reset_index()
+    
+    """Add missing transmission pathways following the approach to disaggregate large GTD
+    regions (e.g. MISO, PJM, SWPP). Transmission pathways between OG regions that fall 
+    within a single GTD region (e.g. USASN and USASS in MISO) are not automatically 
+    pulled from the GTD datasets so need to be manually added. Furthermore, pathways
+    within the GTD datasets that are not selected as the pathways to which residual
+    capacity should be allocated (defined in the CUSTOM_TRN_BA_DICT_FROM and
+    CUSTOM_TRN_BA_DICT_TO constants) also need to be manually added."""
+    df_missing = pd.DataFrame(CUSTOM_TRN_BA_MISSING, columns = ['TECHNOLOGY'])
+    df_missing['From'] = df_missing['TECHNOLOGY'].str[3:8]
+    df_missing['To'] = df_missing['TECHNOLOGY'].str[8:13]
+    df_missing['YEAR'] = '-'
+    df_missing['VALUE'] = 0
```

Comment:
Does PLEXOS-World regions follow [NARIS/ReEDS boundaries](https://www.nrel.gov/analysis/naris.html)? If so, we can just grab transmission data from there and not worry about doing this custom mapping.

Reply:
[Figure 4](https://www.nrel.gov/docs/fy21osti/78195.pdf) shows spatial structure

Reply:
No it mostly follows NEMS see Figure 3 [https://www.eia.gov/outlooks/aeo/nems/documentation/electricity/pdf/EMM_2022.pdf](url). I think that when we start incorporating GEM data later this year/early next year we should reconsider the US nodes within OG. I.e. if we have shapefiles/boundaries we can allocate individual plants within GEM more accurately.